sound/soc/sof/sof-audio.h

```diff
@@ -151,6 +152,8 @@ int snd_sof_complete_pipeline(struct device *dev,
 int sof_load_pipeline_ipc(struct device *dev,
 			  struct sof_ipc_pipe_new *pipeline,
 			  struct sof_ipc_comp_reply *r);
+int sof_pipeline_core_enable(struct snd_sof_dev *sdev,
+			     const struct snd_sof_widget *swidget);
```

Comment:
ok, it's the end of my day and my brain is fried, so maybe a stupid question but shouldn't we have a symmetric sof_pipeline_core_disable()? Or if it's handled automagically add a comment maybe?

Reply:
@plbossart I don't think this is currently supported. DSP cores are enabled when pipelines, that should run on them are loaded, not when they are used. And we don't load or unload pipelines dynamically so far. So, as soon as a topology with pipelines, running on slave cores is loaded, those cores will be enabled and they'll stay enabled until the DSP is turned off.

Reply:
oh wow, that's a really big issue then. we should really plan for cores to be turned off when there's nothing scheduled on them. 

@lgirdwood need you to chime in...

Reply:
@plbossart possibly, but this is really out of the scope of this PR. Note, that this PR doesn't *add* this functionality, it just extends it to the component level.

Reply:
The ask was to support multi-core usages, but not to leave the cores on at all times, even when unused. Maybe that's a change of scope or an undocumented capability, but do you disagree this is needed?

Reply:
@plbossart I don't disagree with that. This PR isn't by any means a complete multi-core support. This is just the first step. After it we can decide which further steps we want to make and how. For this feature - powering down slave cores when not in use - I think we'll need something like or indeed per-core runtime PM. Powering up a core isn't as simple as powering up a simple fixed-function device. When powered up slave cores execute their iniitialisation code, allocate memory, interrupts, initiialise scheduling etc. Also the notion of a "core in use" might be not so simple. E.g. as far as I understand the current plan is to support for example pipelines, running on the master core, where just one or more components from it run on slave cores. So, each pipeline would have a bit-mask of used cores. All this is future development.

Reply:
ok. do you mind adding comments on current limitations, both in code and commit messages, so that upstream reviewers are not surprised when the second batch comes their way?

@lgirdwood @mmaka1 @lbetlej we need to have a discussion on this, there can be multiple ways of dealing with core power management.

Reply:
@plbossart done